### ISSUE TITLE
rotate zoom recording

### ISSUE THREAD

**Issue by alg520** [niqdev/ipcam-view#71]
I tried to implement the rotation and zoom function based on this [example,](https://github.com/google/grafika/blob/master/app/src/main/java/com/android/grafika/TextureFromCameraActivity.java) but failed. Do you have any good Suggestions? There is also a sample in this project for [recording.](https://github.com/google/grafika/blob/master/app/src/main/java/com/android/grafika/VideoEncoderCore.java)

**Comment by niqdev** [comment]
Hi, thanks for sharing the project [grafika](https://github.com/google/grafika)!

For rotation you mean #68?
For zoom you could create a PR out of #55 if you want.
About recording have a look at #10 for ideas.

Unfortunately, I haven't that many suggestions other than dig into [MjpegViewDefault](https://github.com/niqdev/ipcam-view/blob/master/mjpeg-view/src/main/java/com/github/niqdev/mjpeg/MjpegViewDefault.java) and [MjpegInputStreamDefault](https://github.com/niqdev/ipcam-view/blob/master/mjpeg-view/src/main/java/com/github/niqdev/mjpeg/MjpegInputStreamDefault.java) and figure out what they do, I've just nicely wrapped them.

If you think is worth it cos is another complementary implementation, since isn't a lib, you could extract the part that you are interested in and make it extend the base classes and end up with something like

```java
<com.github.niqdev.mjpeg.MjpegSurfaceView
      android:id="@+id/VIEW_NAME"
      android:layout_width="match_parent"
      android:layout_height="match_parent"
      stream:type="stream_grafika" />
```

**Comment by emrekilic** [comment]
hello @alg520.
Have you been able to rotate mjpeg images. My source video images are 90 degrees inverted. The flip feature does not fix it. Do you have any suggestion to rotate the images 90 degrees?

**Comment by xeno14** [comment]
Hi. I submitted #75 to implement image rotation feature. Please take a look.

**Comment by niqdev** [comment]
Hi @xeno14 , thanks for your PR!

**Comment by niqdev** [comment]
@emrekilic  I've released `v1.6.0` of the library and published the latest version of the app on the store.
Feel free to reopen the issue if there is any problem.
Thanks @xeno14

**Comment by emrekilic** [comment]
@xeno14 and @niqdev 
Thanks for your efforts. You helped very quickly.
